### ISSUE TITLE
cifar10 example diverges

### ISSUE THREAD

**Issue by sephirothvs** [BVLC/caffe#1105]
0918 02:05:55.594841  5721 net.cpp:125] loss needs backward computation.
I0918 02:05:55.594844  5721 net.cpp:156] This network produces output accuracy
I0918 02:05:55.594849  5721 net.cpp:156] This network produces output loss
I0918 02:05:55.594866  5721 net.cpp:402] Collecting Learning Rate and Weight Decay.
I0918 02:05:55.594872  5721 net.cpp:167] Network initialization done.
I0918 02:05:55.594877  5721 net.cpp:168] Memory required for data: 0
I0918 02:05:55.594929  5721 solver.cpp:46] Solver scaffolding done.
I0918 02:05:55.594935  5721 solver.cpp:165] Solving CIFAR10_full
I0918 02:05:55.594948  5721 solver.cpp:232] Iteration 0, Testing net (#0)
I0918 02:06:02.779670  5721 solver.cpp:270] Test score #0: 0.1
I0918 02:06:02.779747  5721 solver.cpp:270] Test score #1: 2.30258
I0918 02:06:02.857022  5721 solver.cpp:195] Iteration 0, loss = 2.30264
I0918 02:06:02.857064  5721 solver.cpp:365] Iteration 0, lr = 0.001
I0918 02:06:38.561959  5721 solver.cpp:195] Iteration 200, loss = nan
I0918 02:06:38.562106  5721 solver.cpp:365] Iteration 200, lr = 0.001
I0918 02:07:29.509574  5721 solver.cpp:195] Iteration 400, loss = nan
I0918 02:07:29.509680  5721 solver.cpp:365] Iteration 400, lr = 0.001
I0918 02:08:01.940896  5721 solver.cpp:195] Iteration 600, loss = nan
I0918 02:08:01.941033  5721 solver.cpp:365] Iteration 600, lr = 0.001
I0918 02:08:53.168071  5721 solver.cpp:195] Iteration 800, loss = nan
I0918 02:08:53.168197  5721 solver.cpp:365] Iteration 800, lr = 0.001
I0918 02:09:24.503336  5721 solver.cpp:232] Iteration 1000, Testing net (#0)
I0918 02:09:35.667853  5721 solver.cpp:270] Test score #0: 0.1
I0918 02:09:35.667939  5721 solver.cpp:270] Test score #1: nan
I0918 02:09:36.034212  5721 solver.cpp:195] Iteration 1000, loss = nan
I0918 02:09:36.034260  5721 solver.cpp:365] Iteration 1000, lr = 0.001

I have tried caffe for cifar10 with the example model ./train_full.sh and I have not change anything. But strangely it seems going wrong from the beginning. Any suggestions? The gpu I use is NVIDIA TITAN

while for the fast case ./train_quick.sh it works fine 
I0918 01:58:54.916841 14228 solver.cpp:294] Snapshotting solver state to cifar10_quick_iter_5000.solverstate
I0918 01:58:54.956882 14228 solver.cpp:213] Iteration 5000, loss = 0.572535
I0918 01:58:54.956934 14228 solver.cpp:232] Iteration 5000, Testing net (#0)
I0918 01:58:58.008656 14228 solver.cpp:270] Test score #0: 0.758
I0918 01:58:58.008703 14228 solver.cpp:270] Test score #1: 0.736689
I0918 01:58:58.008712 14228 solver.cpp:218] Optimization Done.
I0918 01:58:58.008718 14228 caffe.cpp:113] Optimization Done.

So where goes wrong to my case, thanks in advance. 
- Wenlin
